### PR TITLE
Feature/input system

### PR DESCRIPTION
**Pull request by PhoebeMitchell**
## Summary

Adds the Input System package, along with code to create an Input Actions object (PlayerInput.cs)
Unity's new Input System will make supporting other devices easier (gamepads/mobile), and allow controls to be configurable at runtime.
Also updates the .gitignore to ignore Rider's cache directory.

## Testing instructions

Check scenes with event systems run as expected. Event systems may need to be updated to use the new input system.

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:

### CODE REVIEW

**Comment by notexplosive** [comment]
How do I review this? I don't really see any code changes? I'm inclined to just say "yeah that's probably right" and hit approve.

**Comment by lyndsiWilliams** [comment]
> How do I review this? I don't really see any code changes? I'm inclined to just say "yeah that's probably right" and hit approve.

I think that's fine, there won't always be something wrong. Just your stamp of "Hey I looked through this and it looks good to me" is a good review, it still shows that another pair of eyes checked the code and _didn't_ find anything 😁

File: Assets/Scripts/Input/Controls.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    ""controlSchemes"": [
+        {
+            ""name"": ""Mouse and Keyboard"",
+            ""bindingGroup"": ""Mouse and Keyboard"",
+            ""devices"": []
+        }
+    ]
+}");
+        // Player
+        m_Player = asset.FindActionMap("Player", throwIfNotFound: true);
```
Comment (lyndsiWilliams) [inline Assets/Scripts/Input/Controls.cs:54]: Member variables should start with just `_`, the m should be removed. [Code style doc reference](https://docs.google.com/document/d/1zN4Yx62PpyXhu1g_AhtTZSC8k9e4zqWbzQPt1Hlw9Ag/edit#heading=h.bhrnxu6997n8)

**Review by lyndsiWilliams — CHANGES_REQUESTED** [review]
Just a code style nit 😁

File: Assets/Scripts/Input/Controls.cs (hunk 1 above)
Comment (bas-mentjot) [reply Assets/Scripts/Input/Controls.cs:54]: This is an auto generated script, best to leave it alone

Comment (PhoebeMitchell) [reply Assets/Scripts/Input/Controls.cs:54]: This class is generated by Input System, so I think its best to leave it as is.

**Comment by PhoebeMitchell** [comment]
> How do I review this? I don't really see any code changes? I'm inclined to just say "yeah that's probably right" and hit approve.

It just adds the Input System package, so pretty much just check if there are any errors.
PlayerInput is the only script I created.

File: Assets/Scripts/Input/Controls.cs (hunk 1 above)
Comment (lyndsiWilliams) [reply Assets/Scripts/Input/Controls.cs:54]: Oooo are the `m`s automatic? That could get confusing, maybe we should revisit the `m_` naming rule

**Review by lyndsiWilliams — APPROVED** [review]

**Comment by PhoebeMitchell** [comment]
Closed because it is now part of feature/Main-Menu

**Pull request closed without merging**